src/metadata_generator.py

```diff
@@ -579,7 +582,8 @@ def update_outputs(
         - Sets `mass_spec_obj.has_output` to [raw_data_obj.id].
         - Sets `analysis_obj.has_output` to `processed_data_id_list`.
         """
-        mass_spec_obj.has_output = [raw_data_obj.id]
+        if not rerun:
```

Comment:
Could you add a comment here (or in the docstring) explaining why the raw_data_object_id is only populated in the non-re-run scenario?